dotcom-rendering/scripts/dotenv/gen-dotenv.js

```diff
@@ -11,77 +11,100 @@ const checkEnv = async () => {
 	try {
 		const env = (await fs.readFile(ENV_PATH)).toString();
 
-		let valid = true;
 		for (const secret of secrets) {
-			const regex = new RegExp(`^${secret.key.replace('.', '\\.')}=.*`);
-			if (!env.match(regex)) valid = false;
+			const regex = new RegExp(`^${secret.key.replace('.', '\\.')}=.+`);
```

Comment:
Why the double-backslash in the `replace`?

Maybe outside the scope of this PR, but it would be nice to always have `RegExp` that are documented and testable. We could also circumvent the regex by splitting the string on `\n` and  `=`.

```ts
/*
KEY=value
OTHER_KEY=other_value
*/
const content = s.readFileSync('file.txt').toString().split("\n");
const entries = content.filter(line => line.trim() !== "").map(line => line.split("="));
const env = Object.fromEntries(entries);
/*
{
  "KEY": "value",
  "OTHER_KEY": "other_value"
} 
*/
```

Reply:
> Why the double-backslash in the replace

I believe this is required because of the weird way that the regex is getting added as a string? 

>  could also circumvent the regex by splitting the string on \n and =.

I'd agree this is the nicer solution so I'll give this a shot just now :)

Reply:
And you can then check for the existence of a key using `'key' in obj`

Reply:
> And you can then check for the existence of a key using 'key' in obj

We actually want to check that both the key exists and is not empty (e.g `IMAGE_SALT=` should error)